app/grandchallenge/notifications/tasks.py

```diff
@@ -0,0 +1,40 @@
+from celery import shared_task
+from django.core.paginator import Paginator
+from django.utils.timezone import now
+
+from grandchallenge.notifications.emails import send_unread_notifications_email
+from grandchallenge.notifications.models import Notification
+from grandchallenge.profiles.models import UserProfile
+
+
+@shared_task
+def send_unread_notification_emails():
+    profiles = UserProfile.objects.filter(
+        receive_notification_emails=True
+    ).order_by("user")
+    paginator = Paginator(profiles, 1000)
+
+    for page_nr in paginator.page_range:
+        current_page_profiles = paginator.page(page_nr).object_list
+        current_time = now()
+        recipients = {}
+        for profile in current_page_profiles:
+            if (
+                profile.notification_email_last_sent_at is None
+                and profile.has_unread_notifications
+            ):
+                recipients[profile] = profile.unread_notifications.count()
+                profile.notification_email_last_sent_at = current_time
+                profile.save()

```

Comment:
Rather than calling `profile.save()` each time, at the end of the for loop you can do `Notification.objects.bulk_update(current_page_profiles, ["notification_email_last_sent_at"])` and that gets it done in 1 database call.